addons/controllers/calico/calicoconfig_controller.go

```diff
@@ -27,15 +45,39 @@ type CalicoConfigReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CalicoConfig object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
 func (r *CalicoConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("calicoconfig", req.NamespacedName)
+	ctxCancel, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+	log := r.Log.WithValues("CalicoConfig", req.NamespacedName)
+
+	log.Info("Start reconciliation")
+
+	// fetch CalicoConfig resource, ignore not-found errors
+	calicoConfig := &cniv1alpha1.CalicoConfig{}
+	if err := r.Client.Get(ctxCancel, req.NamespacedName, calicoConfig); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
question: would it be a good idea to do some logging here ? For example, if `calicoConfig` is not found, we would have a better idea on debugging.

Reply:
Agreed, Done.

Reply:
suggestion: usually we will do a deep copy here and pass down the copied `calicoConfig`. So that you do not need to do multiple deep copy at L109 and L115